src/group_theory/sylow.lean

```diff
@@ -432,4 +432,77 @@ begin
   rwa [h, card_bot] at key,
 end
 
+/- A sylow subgroup in G is also a sylow subgroup in a subgroup of G. -/
+def subtype {p : ℕ} (P : sylow p G) (N : subgroup G) (h : ↑P ≤ N) :
+  sylow p N :=
+   ⟨ comap N.subtype P.1,
+     is_p_group.comap_subtype P.is_p_group',
+     begin
+       intros Q hQ hle,
+       let Q' : subgroup G := map N.subtype Q,
+       have h' : P.to_subgroup ≤ N.subtype.range := by simpa,
+       have hQ' : is_p_group p Q' := is_p_group.map hQ _,
+       have hle' : P.to_subgroup ≤ Q':= calc P.to_subgroup
+              = map N.subtype (comap N.subtype P.to_subgroup) : symm $ map_comap_eq_self h'
+          ... ≤ Q' : map_mono hle,
+       rewrite ← P.is_maximal' hQ' hle',
+       symmetry,
+       apply comap_map_eq_self_of_injective,
+       exact subtype.val_injective,
+     end ⟩
+
+@[simp]
+lemma coe_subtype {p : ℕ} {P : sylow p G} {N : subgroup G} {h : P.1 ≤ N} :
+  ↑(P.subtype N h) = comap N.subtype ↑P := rfl
+
+lemma normal_of_normalizer_normal [fintype G] {p : ℕ} (hp : fact p.prime) (P : sylow p G)
+  (hn : (↑P : subgroup G).normalizer.normal) :
+  (↑P : subgroup G).normal :=
+begin
+  let N := (↑P : subgroup G).normalizer,
+  let P' : sylow p N := P.subtype N le_normalizer,
+
+  -- NB: This lemma assumes (fintype G), and not merely fintype (sylow p G).
+  -- The latter would suffices if we could deduce (sylow p N) from it.
+
+  have := normalizer_sup_eq_top P',
+  simp [N, P'] at this,
+  rewrite map_comap_eq_self at this,
+  show (_ ≤ _), { simp only [subtype_range], apply le_normalizer, },
+  apply normalizer_eq_top.mp,
+  rw sup_idem at this,
+  apply this,
+end
+
+lemma normalizer_self_normalizing [fintype G] {p : ℕ} {hp : fact p.prime} (P : sylow p G) :
+ (↑P : subgroup G).normalizer.normalizer = (↑P : subgroup G).normalizer :=
+begin
+  let H := (↑P : subgroup G).normalizer.normalizer,
+  show (H = (↑P : subgroup G).normalizer),
+  let P' : sylow p H := P.subtype H (le_trans le_normalizer le_normalizer),
+
+  -- NB: This lemma assumes (fintype G), and not merely fintype (sylow p G).
+  -- The latter would suffices if we could deduce (sylow p N) from it.
+
+  have h1 : (P' : subgroup H).normalizer = ⊤ :=
+  begin
+    apply normalizer_eq_top.mpr,
+    apply normal_of_normalizer_normal hp P',
+    apply normalizer_eq_top.mp,
+    simp [P'],
+    rewrite ← comap_subtype_normalizer_eq le_normalizer,
+    rewrite ← comap_subtype_normalizer_eq (le_refl _),
+    rewrite subgroup.comap_subtype_self_top,
+  end,
```

Comment:
```suggestion
  have h1 : (P' : subgroup H).normalizer = ⊤,
  { apply normalizer_eq_top.mpr,
    apply normal_of_normalizer_normal hp P',
    apply normalizer_eq_top.mp,
    simp [P'],
    rewrite ← comap_subtype_normalizer_eq le_normalizer,
    rewrite ← comap_subtype_normalizer_eq (le_refl _),
    rewrite subgroup.comap_subtype_self_top, },
```
we tend to use this style more as it avoids opening nested tactic blocks, rather just manipulating the focussed goal within one tactic block

Reply:
Thanks. Is there a technical difference, or just aesthetics?

Reply:
My understanding is that its not just cosmetic. I'm on mobile so can't get the link but hopefully this screenshot of when I was first told this is helpful ![Screenshot_20220124-183058.png](https://user-images.githubusercontent.com/1688533/150834644-b063e052-6de8-42d2-adfd-4b2a06d1dd5a.png)

Reply:
Thanks for digging that up!